### DISCUSSION TITLE
Will there be a native support for Win32 in the next version of Windows (Windows 11)?

### DISCUSSION THREAD

**Discussion by zaffinger** [microsoft/WindowsAppSDK#934] [Q&A]
The title says it all.

It sucks to be a Delphi developer looking at this [image](https://heise.cloudimg.io/width/1920/q85.png-lossy-85.webp-lossy-85.foil1/_www-heise-de_/imgs/18/2/8/9/7/5/9/2/image1-bf65157fa96a124d.png).
source: [heise.de](https://www.heise.de/developer/meldung/Microsoft-Build-2020-Microsoft-will-Windows-APIs-wiedervereinen-4724998.html)

If the question is completely wrong, then I apologize.

**Comment by mdtauk** [comment]
~Yes~

Edit: We can't know for sure until either the June 24th event, or until the OS is released to the public.  At the moment there does not appear to be anything put in place that would block native Win32 apps from running.  Project Reunion states that it plans to be an opt in, and you don't have to abandon your existing apps.  If that is anything to go by.

Sorry for my earlier flippant reply, I took the humour of your Delphi comment as a not super serious question being asked.  Thanks @riverar for pulling me up on this.

**Reply by riverar** [reply]
@mdtauk You don't know that.

**Reply by mdtauk** [reply]
> @mdtauk You don't know that.
>
You are correct, I should couch the comment in some ambiguity.

**Comment by zaffinger** [comment]
And what about all the fancy UWP stuff? Will there be (finally) native Win32 support for this in the new Windows?

**Reply by mdtauk** [reply]
> And what about all the fancy UWP stuff? Will there be (finally) native Win32 support for this in the new Windows?

Project Reunion and WinUI 3.0 is all about bringing support for the modern UI to Win32 apps.  You should keep an eye on the roadmaps for both, for what limitations are currently in place, and for when certain features and functionality will be in place and ready to use.

**Comment by just1a-person** [comment]
Yes

Source: Windows Event

**Reply by amartens181** [reply]
Are we sure it's native support and not a VM?

**Reply by just1a-person** [reply]
Win32 is native.
